stac_fastapi/extensions/stac_fastapi/extensions/core/filter/filter.py

```diff
@@ -124,3 +125,112 @@ def register(self, app: FastAPI) -> None:
+            FilterConformanceClasses.CQL2_TEXT,
+        ]
+    )
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        self.router.prefix = app.state.router_prefix
+        self.router.add_api_route(
+            name="Queryables",
+            path="/queryables",
+            methods=["GET"],
+            responses={
+                200: {
+                    "content": {
+                        "application/schema+json": {},
+                    },
+                    # TODO: add output model in stac-pydantic
+                },
+            },
+            response_class=self.response_class,
+            endpoint=create_async_endpoint(self.client.get_queryables, EmptyRequest),
+        )
+        app.include_router(self.router, tags=["Filter Extension"])
+
+
+@attr.s
+class ItemCollectionFilterExtension(FilterExtension):
+    """Item Collection Filter Extension."""
+
+    conformance_classes: List[str] = attr.ib(
+        default=[
+            FilterConformanceClasses.FILTER,
+            FilterConformanceClasses.ITEMS,
+            FilterConformanceClasses.BASIC_CQL2,
+            FilterConformanceClasses.CQL2_JSON,
+            FilterConformanceClasses.CQL2_TEXT,
+        ]
+    )
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        self.router.add_api_route(
+            name="Collection Queryables",
+            path="/collections/{collection_id}/queryables",
+            methods=["GET"],
+            responses={
+                200: {
+                    "content": {
+                        "application/schema+json": {},
+                    },
+                    # TODO: add output model in stac-pydantic
+                },
+            },
+            response_class=self.response_class,
+            endpoint=create_async_endpoint(self.client.get_queryables, CollectionUri),
+        )
+        app.include_router(self.router, tags=["Filter Extension"])
+
+
+@attr.s
+class CollectionSearchFilterExtension(FilterExtension):
+    """Collection Search Filter Extension."""
+
+    conformance_classes: List[str] = attr.ib(
+        default=[
+            FilterConformanceClasses.FILTER,
+            FilterConformanceClasses.COLLECTIONS,
+            FilterConformanceClasses.BASIC_CQL2,
+            FilterConformanceClasses.CQL2_JSON,
+            FilterConformanceClasses.CQL2_TEXT,
+        ]
+    )
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        pass
```

Comment:
maybe worth for add a note about why we're not registering default endpoint here

ref: https://github.com/stac-utils/stac-fastapi-pgstac/pull/192#discussion_r1927373445